src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs

```diff
@@ -15,30 +18,50 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
     /// </summary>
     internal class InlineParameterNameHintsTag : IntraTextAdornmentTag
     {
+        public const string TagId = "inline parameter name hints";
+
         /// <summary>
         /// Creates the UIElement on call
         /// Uses PositionAffinity.Successor because we want the tag to be associated with the following character
         /// </summary>
         /// <param name="text">The name of the parameter associated with the argument</param>
-        public InlineParameterNameHintsTag(string text)
-            : base(CreateElement(text), removalCallback: null, PositionAffinity.Successor)
+        public InlineParameterNameHintsTag(string text, double lineHeight, TextFormattingRunProperties format)
+            : base(CreateElement(text, lineHeight, format), removalCallback: null, PositionAffinity.Successor)
         {
         }
 
-        private static UIElement CreateElement(string text)
+        private static UIElement CreateElement(string text, double lineHeight, TextFormattingRunProperties format)
         {
-            // Constructs the hint block which gets assigned parameter name, a normal fontstyle, and sets the padding 
-            // space around the block to 0
+            // Constructs the hint block which gets assigned parameter name and nested inside a border where the 
+            // attributes of the block are received from the options menu
             var block = new TextBlock
             {
-                Text = text + ": ",
+                Text = text + ":",
+                HorizontalAlignment = HorizontalAlignment.Center,
+                VerticalAlignment = VerticalAlignment.Center,
                 FontStyle = FontStyles.Normal,
+                FontFamily = format.Typeface.FontFamily,
+                FontSize = format.FontRenderingEmSize - 1,
                 Padding = new Thickness(0),
+                Background = format.BackgroundBrush,
+                Foreground = format.ForegroundBrush
             };
 
-            block.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
-
-            return block;
+            var border = new Border
+            {
+                CornerRadius = new CornerRadius(2),
```

Comment:
Want to sort this list of properties? It doesn't really matter but can make it a bit easier to find one in the list.